src/ring_theory/adjoin/power_basis.lean

```diff
@@ -62,3 +62,11 @@ where `d` is the degree of the minimal polynomial of `x`. -/
   basis_eq_pow := basis.mk_apply _ _ }
 
 end algebra
+
+open algebra
+
+/-- The power basis given by `x` if `B.gen ∈ adjoin K {x}`. -/
+@[simps gen dim] noncomputable def power_basis.of_gen_mem_adjoin {x : S} (B : power_basis K S)
```

Comment:
I thought we set up `simps` so that `gen dim` is the default

Reply:
You're right.